pkg/sql/alter_database.go

```diff
@@ -358,6 +360,24 @@ func (p *planner) AlterDatabaseDropRegion(
 	}, nil
 }
 
+// ensureCorrectMultiRegionPrivileges ensures the current user has the required
+// privileges to alter the locality configuration of the given table descriptor.
+// Currently, this entails that the user must have either CREATE or ZONECONFIG
+// privilege on the table descriptor.
+// TODO(arul): This ought to be unified with `ALTER TABLE LOCALITY` as well.
+func ensureCorrectMultiRegionPrivileges(
+	ctx context.Context, p *planner, tableDesc catalog.TableDescriptor,
+) error {
+	zoneConfigPrivErr := p.CheckPrivilege(ctx, tableDesc, privilege.ZONECONFIG)
+	createPrivErr := p.CheckPrivilege(ctx, tableDesc, privilege.CREATE)
+	if zoneConfigPrivErr != nil && createPrivErr != nil {
```

Comment:
Since we eventually want to be more accurate with Postgres privileges and they don't have `CREATE` privilege for tables, maybe we should only show in the error that the user doesn't have `ZONECONFIG`.  Or maybe we should show the same error message for ALTER TABLE LOCALITY `must be owner of table %s or have CREATE privilege on table %s`

Actually would it make more sense to pull out the privilege check in `alter_table_locality.go` (and possibly even `alter_table.go` into a helper function and just re-use it here so we don't duplicate privilege checks that should technically be the same?

Reply:
> Since we eventually want to be more accurate with Postgres privileges and they don't have CREATE privilege for tables, maybe we should only show in the error that the user doesn't have ZONECONFIG. 

If we're accepting CREATE as a valid privilege to do this operation, then IMO the error message should indicate that. If we want to move more in line with Postgres then maybe we shouldn't accept CREATE here at all to begin with. Would you prefer that?

> Or maybe we should show the same error message for ALTER TABLE LOCALITY must be owner of table %s or have CREATE privilege on table %s

I think that thing is busted to be honest, it should accept ZONECONFIG as a valid privilege too. I was thinking of calling this function in `alter_table_locality.go`,  there's a TODO, I'll just address it here. That being said, I think it makes sense to add a nod to the "owner" part in the error message. 

> Actually would it make more sense to pull out the privilege check in alter_table_locality.go (and possibly even alter_table.go into a helper function and just re-use it here so we don't duplicate privilege checks that should technically be the same?

But aren't the privilege checks different for `alter_table_locality.go` and `alter_table.go`? IMO `alter_table_locality.go` should accept ZONECONFIG but `alter_table.go` should just be CREATE.

Reply:
I was thinking we could pull out some of the shared logic for the error check from `alter_table.go` and `alter_table_locality.go` and use it in all three places. Something like one helper function for checking `CREATE` or `OWNER` which is used in all 3, however for `alter_table_locality.go` and this function here, we can wrap that helper and add an additional check for `ZONECONFIG`. If this turns out to be messy then I'd be okay with just creating one function for the locality checks.

Reply:
> there's a TODO, I'll just address it here. That being said, I think it makes sense to add a nod to the "owner" part in the error message.

Changed my mind, I'll address this in a separate commit. It needs some testing as well.

Reply:
Ah, I see what you mean, I'm on board. I'll do it when I address the `alter_table_locality.go` stuff :+1:

Reply:
Do we have an issue for the `alter_table_locality.go` privilege stuff or a general issue on the MR privilege stuff?

Reply:
I did the alter_table_locality in the second commit. As for other MR privileges stuff, I still have to write them out after the whole Slack discussion.

Reply:
Followups for after this thing goes in: 

https://github.com/cockroachdb/cockroach/issues/62252
https://github.com/cockroachdb/cockroach/issues/62250